server/api-manager.js

```diff
@@ -968,9 +971,26 @@ class MindbodyQueries {
     request.addAuth(this.authToken);
     this.requestNum++;
     let allPagePromises = [];
+
+    let makeRequest = () => {
+      return request.makeRequest();
+    };
+    //backoff function, backs off a preset amount
+    const backoff = (retries, fn, delay = this.initialDelay) => {
+      return fn().catch((err) => {
+        if (retries > 1 && !this.atLimit()) {
```

Comment:
When I did this in the other branch, I separated out the !atLimit from retries:
        if (this.atLimit()) {
          return Promise.reject(Error("Mindbody request limit reached"));
        }
        if (retries > 1) {
...I think it reads more clearly. Also, this code generates bad console logs in the (potentially rare) case where we run out of Mindbody requests at this code path. Instead of throwing a "Mindbody request limit reached" Error, we'll throw the error message for whatever made the previous request fail. But that error message is almost certainly less important to users than 'Mindbody request limit reached"

Reply:
Good catch